src/pipeline_migration/actions/migrate/cli.py

```diff
@@ -35,15 +111,117 @@ def register_cli(subparser) -> None:
         "-l",
         "--use-legacy-resolver",
         action="store_true",
         help="Use legacy resolver to fetch migrations.",
     )
+    migrate_parser.add_argument(
+        "-p",
+        "--pipeline-file",
+        metavar="PATH",
+        dest="pipeline_files",
+        action="append",
+        type=arg_type_pipeline_file,
+        help="Relative path to Pipeline/PipelineRun YAML file. Used in conjunction with option "
+        "-b/--new-bundle only. If omitted, pipeline files will be searched from YAML files "
+        "under ./.tekton/ directory that have kind Pipeline or PipelineRun.",
+    )
     migrate_parser.set_defaults(action=action)
 
 
+class DotTekton(Path):
+
+    def list_pipeline_files(self) -> Iterable[Path]:
+        for possible_yaml_file in self.glob("*.y[a]ml"):
+            doc = None
+            with suppress(ScannerError, IOError):
+                doc = YAML().load(possible_yaml_file.read_text())
+            if doc and isinstance(doc, dict):
+                kind = doc.get("kind")
+                if kind == "Pipeline" or kind == "PipelineRun":
+                    yield possible_yaml_file
+
+
+def search_pipeline_files() -> list[str]:
+    """Search pipeline files from .tekton/
+
+    :param dir_path: A directory path from where to search pipeline files. Pipeline files are Tekton
+        Pipeline/PipelineRun YAML files.
+    :type dir_path: str
+    :return: A list of pipeline file paths.
+    :rtype: list[str]
+    """
+    tekton_dir = DotTekton("./.tekton")
+    if not tekton_dir.exists():
+        logger.info("Current working directory does not have directory .tekton/.")
+        return []
+    pipeline_files = list(tekton_dir.list_pipeline_files())
+    if not pipeline_files:
+        logger.info("No Tekton Pipeline/PipelineRun is found from directory .tekton/.")
+        return []
+    return list(map(str, pipeline_files))
+
+
+def generate_upgrades_data(new_bundles: list[str], pipeline_files: list[str]) -> str:
+    """Generate Renovate upgrades
+
+    :param new_bundles: a list of new bundle references. Each of them must have both tag and digest.
+    :type: list[str]
+    :param pipeline_files: a list of pipeline files. Upgrades will be generated for each of these
+        pipeline files.
+    :type pipeline_files: list[str]
+    :return: a string representing the upgrades. That includes fields Renovate template field
+        ``upgrades`` has.
+    :rtype: str
+    """
+    upgrades: list[RenovateUpgradeT] = []
+    for pipeline_file in pipeline_files:
+        with open(pipeline_file, "r") as f:
+            content = f.read()
+        for bundle_ref in new_bundles:
+            new_c = Container(bundle_ref)
+            dep_name = f"{new_c.registry}/{new_c.api_prefix}"
+            regex = REGEX_BUNDLE_REF_VALUE.replace("dep_name", dep_name)
+            if match := re.search(regex, content):
+                _, current_bundle_ref, current_tag, _, current_digest = match.groups()
```

Comment:
Is it OK to ignore the 4th capture group? Shouldn't it be part of `current_tag`?